src/components/Modal/Modal.js

```diff
@@ -0,0 +1,145 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { BsArrowRight } from 'react-icons/bs';
+import { GrClose } from 'react-icons/gr';
+import { useRecoilState, useRecoilValue } from 'recoil';
+import showModalState, {
+  selectedDatesState,
+  selectedLocationState,
+  validDatesState,
+} from './GlobalState';
+import Location from '../Modal/Location/Location';
+import SelectDate from './SelectDate/SelectDate';
+import { useEffect } from 'react/cjs/react.development';
+import convertToQs from '../QueryString/QueryString';
+import { useNavigate } from 'react-router';
+
+export default function StaticDateRangePickerDemo() {
+  const navigate = useNavigate();
+  const [showModal, setShowModal] = useRecoilState(showModalState);
+  const selectedDates = useRecoilValue(selectedDatesState);
+  const selectedLocation = useRecoilValue(selectedLocationState);
+  const validDates = useRecoilValue(validDatesState);
+  const [buttonIsValid, setButtonIsValid] = React.useState(false);
+
+  function onClickSearch() {
+    setShowModal(null);
+    if (showModal === 'location') {
+      navigate(convertToQs('list', selectedLocation));
+    } else if (showModal === 'date') {
+      navigate(convertToQs('list', selectedDates));
+    } else if (showModal === 'date_detail') {
+      console.log(selectedDates);
+    }
+  }
+
+  useEffect(() => {
+    function disableButton() {
+      if (showModal === 'location') {
+        setButtonIsValid(selectedLocation.location !== null);
+      } else if (showModal === 'date' || showModal === 'date_detail') {
+        setButtonIsValid(validDates);
+      }
+    }
+    disableButton();
+  }, [selectedLocation, validDates, selectedDates, showModal, buttonIsValid]);
+
+  function onClose() {
+    setShowModal(null);
+  }
+
+  const modalTitle = {
+    location: '어디로 떠날까요?',
+    date: '언제 떠날까요?',
+    date_detail: '예약 날짜를 선택해주세요.',
+  };
+
+  const modalButton = {
+    location: 'SEARCH',
+    date: 'SEARCH',
+    date_detail: 'DONE',
+  };
```

Comment:
컴포넌트 내 state, props에 영향을 받지 않는 상수나 함수는 불필요한 메모리 할당을 줄이기 위해 컴포넌트 밖에 위치하는 것이 좋습니다. 변하지 않는 상수일 경우, 상수명은 '대문자'로 해주세요!
```suggestion
  const MODAL_TITLE = {
    location: '어디로 떠날까요?',
    date: '언제 떠날까요?',
    date_detail: '예약 날짜를 선택해주세요.',
  };

  const MODAL_BUTTON_TO_STRING = {
    location: 'SEARCH',
    date: 'SEARCH',
    date_detail: 'DONE',
  };
```